javascript/javascript3/week3/script.js

```diff
@@ -0,0 +1,240 @@
+        this.x = x;
+        this.y = y;
+        this.r = r;
+        this.startAngle = startAngle;
+        this.endAngle = endAngle;
+        this.fillColor = fillColor;
+        this.ctx = ctx;
+    }
+
+    draw(){
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.x,this.y);
+        this.ctx.arc(this.x, this.y, this.r, this.startAngle,this.endAngle);
+        this.ctx.fillStyle = this.fillColor;
+        this.ctx.fill();
+    }
+}
+
+//Lets make some art using classes
+const canvas1 = document.getElementById("canvas1");
+const ctx1 = canvas1.getContext("2d");
+const circles = [
+    new Circle(282,214,15,0,360,"green",ctx1),
+    new Circle(125,302,90,0,360,"red",ctx1),
+    new Circle(130,94,40,0,360,"blue",ctx1),
+    new Circle(400,460,200,0,360,"yellow",ctx1),
+    new Circle(456,159,38,0,360,"orange",ctx1)
+]
+
+circles.forEach(c=>{
+    c.draw();
+})
+
+
+
+//Now lets make art
+const canvas2 = document.getElementById("canvas2");
+const ctx2 = canvas2.getContext("2d");
+
+function randomInt(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function getRandomColor() {
+    const letters = '0123456789ABCDEF';
+    let color = '#';
+    for (var i = 0; i < 6; i++) {
+      color += letters[Math.floor(Math.random() * 16)];
+    }
+    return color;
+}
+
+const maxCircles = 1000;
+let counter = 0;
+function intervalCircles(){
+    const interv = setInterval(()=>{
+        counter++;
+        const c = new Circle(randomInt(0,500),randomInt(0,500),randomInt(5,30),0,360,getRandomColor(),ctx2);
+        c.draw();
+        if(counter >= maxCircles){
+            clearInterval(interv);
+        }
+    },100)
+}
+
+intervalCircles();
+
+//Classmates
+const classmates = document.getElementById("classmates");
+
+function renderRepositories(username){
+  const getData = new Promise((resolve,reject) =>{
+    fetch(`https://api.github.com/search/repositories?q=user:${username}`).then((msg)=>{
+      return msg.json();
+    }).then((msg)=>{
+      resolve(msg);
+    }).catch((e)=>{
+      reject(e);
+    });
+  });
+  
+  //Fullname, url and owner
+  const userDiv = document.createElement("div");
+  const repOwner = document.createElement("h2");
+  const br = document.createElement("br");
+  repOwner.innerHTML = "Github User: " + username;
+  userDiv.appendChild(repOwner);
+  classmates.appendChild(userDiv);
+  userDiv.classList.add("user-div");
+  
+  getData.then((data)=>{
+    data.items.forEach((rep)=>{
+      const repDiv = document.createElement("div");
+      userDiv.appendChild(repDiv);
+
+      const repName = document.createElement("p");
+      repName.innerHTML = rep.full_name;
+      repDiv.appendChild(repName);
+
+      const repUrl = document.createElement("a");
+      repUrl.href = rep.url;
+      repUrl.innerHTML = "Link";
+      repDiv.appendChild(repUrl);
+
+    })
+  });
+}
+
+renderRepositories("");
```

Comment:
The task was to use `Promise.all` to fetch the data for at least 3 classmates at the same time.